client/lib/validation.rs

```diff
@@ -164,14 +164,23 @@ pub(crate) fn validate_query_response(
         }
     }
 
-    core::validate_query_proof(
-        &state_root_hash.to_owned().into(),
-        &proofs,
-        key,
-        path,
-        proof_value,
-    )
-    .map_err(Into::into)
+    let state_root_hash: Digest = {
+        let value = object
+            .get(GET_ITEM_RESULT_STATE_ROOT_HASH)
+            .ok_or(ValidateResponseError::ValidateResponseFailedToParse)?;
+
+        let value_str = value
+            .as_str()
+            .ok_or(ValidateResponseError::ValidateResponseFailedToParse)?;
+
+        let value_bytes = hex::decode(value_str)
+            .map_err(|_| ValidateResponseError::ValidateResponseFailedToParse)?;
```

Comment:
Perhaps we could just extend `ValidateResponseError` so we have a variant for this?

ie,

```rust
#[error(transparent)]
HexDecodeError(#[from] hex::decode::Error)
```

or something like that?  But maybe we have some downstream formatting stuff for `ValidateResponserError` this would make more complicated.

Reply:
Yeah I agree, I think it would be clearer